jbake-core/src/main/java/org/jbake/parser/MarkupEngine.java

```diff
@@ -171,6 +171,10 @@ private boolean hasHeader(List<String> contents) {
             return false;
         }
 
+        if (!headerSeparatorDemarcatesHeader(contents)) {
```

Comment:
I think this check obsoletes the previous `hasHeaderSeparatorInContent()`. For a quick fix this solution is good.

Reply:
Maybe we need a better parser implementation when the separated header issue gets tackled. We shouldn't iterate twice over the contents.